articles/test/test_views.py

```diff
@@ -0,0 +1,250 @@
+            is_hidden=False,
+            favorited=False,
+            favoritesCount=0,
+            author=self.user,
+        )
+
+    @property
+    def bearer_token(self) -> Any:
+        login_url = reverse("login")
+        response = self.client.post(
+            login_url,
+            data={"email": self.user.email, "password": self.password},
+            format="json",
+        )
+        token = json.loads(response.content).get("access")  # type: ignore[attr-defined]
+        return {"HTTP_AUTHORIZATION": f"Bearer {token}"}
+
+    @patch(
+        "cloudinary.uploader.upload_resource", return_value=fake.image_url()
+    )
+    def test_create_article(self, upload_resource: None) -> None:
+        data = {
+            "title": fake.texts(nb_texts=2),
+            "description": fake.paragraph(nb_sentences=3),
+            "body": fake.paragraph(nb_sentences=20),
+            "image": sample_image(),
+            "is_hidden": False,
+            "tags": f'["{fake.word()}", "{fake.word()}"]',
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertTrue(upload_resource.called)  # type: ignore[attr-defined]
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Article.objects.count(), 2)
+
+    def test_create_article_without_title(self) -> None:
+        data = {
+            "description": fake.text(),
+            "body": fake.text(),
+            "image": sample_image(),
+            "is_hidden": False,
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(Article.objects.count(), 1)
+
+    def test_create_article_without_tags(self) -> None:
+        data = {
+            "title": fake.text(),
+            "description": fake.text(),
+            "body": fake.text(),
+            "image": sample_image(),
+            "is_hidden": False,
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
assert that the message is tags are missiing

Reply:
ok